proto/sbom/sbom.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+option go_package = "github.com/DataDog/agent-payload/v5/sbom";
+
+package datadog.sbom;
+
+import "google/protobuf/timestamp.proto";
+import "proto/deps/github.com/CycloneDX/specification/schema/bom-1.4.proto";
+
+// SBOMPayload represents the main SBOM payload
+message SBOMPayload {
+  int32               version  = 1;
+  string              host     = 2;
+  string              source   = 3; // use to know the source of the message: agent, other
+  repeated SBOMEntity entities = 4;
+}
+
+message SBOMEntity {
+  SBOMSourceType            type        = 1;
+  string                    id          = 2; // Unique identifier to be able to correlated and "deduplicate" SBOM
+  google.protobuf.Timestamp generatedAt = 3; // the datetime of the SBOM generation
+  repeated string           tags        = 4; // datadog tags that will be added by the agent depending of the SBOMSourceType
+  oneof sbom {
+    cyclonedx.v1_4.Bom cyclonedx = 10; // only cyclonedx will be supported initially but putting it optional will allow us to move to another format later
+    // spdx.SBOM      spdx      = 11; // SPDX can be added later as any other format.
+  }
+}
+
+enum SBOMSourceType {
+  UNSPECIFIED            = 0;
```

Comment:
In what cases can it be `UNSPECIFIED`. Can this be useful?

Reply:
It’s just that `0` is the default value when a field isn’t explicitly set or present in the encoded buffer.
Having a dedicated value for `0` allows to make the distinction between a bug where we forgot to set the field and a functionally relevant value.
I basically only mimicked the samples of [the `enum` documentation](https://developers.google.com/protocol-buffers/docs/proto3?hl=en#enum).